libyul/backends/wasm/WasmAST.h

```diff
@@ -30,6 +30,11 @@
 namespace solidity::yul::wasm
 {
 
+using Type = std::string;
```

Comment:
Shouldn't this be an enum instead?

Reply:
Not sure if it should but it definitely could. I based this on `AsmData.h` where `yul::Type` is a `YulString`. Maybe it should be an enum there too?

Anyway, if it's fine with you, I'm changing `wasm::Type` to an enum.

Reply:
In Yul, we always had the concept of user-defined types and dialect can also add new types, so we used a string instead of an enum, but the types for wasm are fixed, so I think it makes sense to use an enum here instead.

Reply:
Oh, OK. I didn't know that user-defined types were already possible in Yul.

Reply:
Done. It's an enum now.